app/signup/page.tsx

```diff
@@ -1,67 +1,60 @@
+import { useCsrf } from '@/context/CsrfContext';
 import logECS from '@/utils/clientLogger';
+import { handleErrors } from '@/utils/errorHandler';
+import { useAuthContext } from '@/context/AuthContext';
 
 const SignUpPage: React.FC = () => {
     const [email, setEmail] = useState("");
     const [password, setPassword] = useState("");
+    const [acceptedTerms, setAcceptedTerms] = useState<number>(0);
     const [errors, setErrors] = useState<string[]>([]);
     const [loading, setLoading] = useState(false);
-    const [attempts, setAttempts] = useState(0);
-    const [lockoutTime, setLockoutTime] = useState<number | null>(null);
     const router = useRouter();
     const errorRef = useRef<HTMLDivElement>(null);
+    const { csrfToken } = useCsrf();
+    const { setIsAuthenticated } = useAuthContext();
 
-    const handleResponse = async (response: Response) => {
-        const { message } = await response.json();
-        if (response.status === 200) {
-            router.push('/') //redirect to home page
-        } else if (response.status === 400) {
-            if (message) {
-                // Convert string into array
-                const errorMessages = message.split('| ');
-                setErrors(prevErrors => {
-                    return [...prevErrors, ...errorMessages]
-                })
-            }
-        } else if (response.status === 500) {
-            logECS('error', 'Internal server error', {
-                url: { path: '/apollo-signup' }
-            });
-            router.push('/500');
-        } else {
-            setErrors(['An unexpected error occurred. Please try again.'])
+    const handleAcceptedTerms = async (event: React.ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+
+        if (event.target.checked) {
+            setAcceptedTerms(1)
         }
-    };
+    }
 
     const handleSignUp = async (event: React.FormEvent) => {
         event.preventDefault();
         setLoading(true);
         setErrors([]); // Clear previous errors
 
-        if (attempts > 5) {
-            const remainingTime = lockoutTime ? Math.max(lockoutTime - Date.now(), 0) : 0;
-            const minutesLeft = Math.ceil(remainingTime / 60000);
-            const remainingMinutesText = minutesLeft < 2 ? 'minute' : 'minutes'
-            setErrors([`Too many attempts. Please try again later${remainingTime > 0 ? ` in ${Math.ceil(remainingTime / 60000)} ${remainingMinutesText}.` : '.'}`]);
-            setLoading(false);
-            return;
-        }
 
-        /* eslint-disable @typescript-eslint/no-explicit-any */
-        try {
-            const response = await fetch(`${process.env.NEXT_PUBLIC_SERVER_ENDPOINT}/apollo-signup`, {
+        const signupRequest = async (token: string | null) => {
+            return await fetch(`${process.env.NEXT_PUBLIC_SERVER_ENDPOINT}/apollo-signup`, {
                 method: 'POST',
+                credentials: 'include',
```

Comment:
Added new `acceptedTerms` field to POST, along with `credentials` and `X-CSRF-TOKEN`.